lsp-mode.el

```diff
@@ -715,6 +715,36 @@ are determined by the index of the element."
                          (const kind)))
   :group 'lsp-imenu)
 
+(defcustom lsp-imenu-index-symbol-kinds '(5 6 7 8 9 10 11 12 23)
```

Comment:
I don't like the fact that the users that don't use customize interface will have to use numbers instead of symbols here. I will change the code that we use for generating stubs to generate vector lookups for enum properties to avoid the need to manually define them and then you could use that to convert number -> symbol or (even generate the defcustom type?).

Reply:
Here it is the PR - https://github.com/emacs-lsp/lsp-mode/pull/2014/